frontend/src/components/common/LoginBtn/RegisterModal/RegisterModal.jsx

```diff
@@ -0,0 +1,168 @@
+      try {
+        const response = await axios.post(
+          'https://api.imperiaholoda.com.ua:4446/v1/auth/register',
+          {
+            firstName: name,
+            email: email,
+            password: password,
+          },
+        );
+        if (response.status === 200) {
+          const token = response.data.token;
+          localStorage.setItem('authToken', token);
+          setRegistrationSuccess(true);
+        }
+      } catch (error) {
+        if (error.response && error.response.status === 400) {
+          setEmailExistsError('Електронна пошта вже зареєстрована!');
+        }
+      }
+    }
+  };
+  return (
+    <>
+      {registrationSuccess ? (
+        <RegisterLetter userEmail={email} toggleModal={toggleRegister} />
+      ) : (
+        <Modals type="login" onClick={toggleRegister}>
+          <div className={styles.header}>
+            <div className={styles.headerContent}>
+              <p className={styles.loginBtn} onClick={toggleLogin}>
+                Вхід
+              </p>
+              <p className={styles.registerBtn}>Реєстрація</p>
+            </div>
+            <div className={styles.closeBtn}>
+              <IconButton
+                icon={<ICONS.closeMobile />}
+                onClick={toggleRegister}
+              />
+            </div>
+          </div>
+
+          <div className={styles.formContainer}>
+            <form onSubmit={handleSubmit}>
+              <div className={styles.dataContainer}>
+                <div className={styles.nameContainer}>
+                  <label className={styles.labelData} htmlFor="name">
+                    Ім’я {submitted && !name && <span>*</span>}
+                  </label>
+
+                  <input
+                    id="name"
+                    className={styles.dataInput}
+                    type="text"
+                    placeholder="Введіть ваше ім’я"
+                    onChange={(e) => setName(e.target.value.trim())}
+                  />
+                  {submitted && !name && (
+                    <p className={styles.errorMessage}>Введіть ваше ім`я</p>
+                  )}
+                </div>
+                <div className={styles.emailContainer}>
+                  <label className={styles.labelData} htmlFor="email">
+                    Ел. пошта {submitted && !email && <span>*</span>}
+                  </label>
+
+                  <input
+                    id="email"
+                    className={styles.dataInput}
+                    type="email"
+                    placeholder="Введіть електронну пошту"
+                    value={email}
+                    onChange={(e) => {
+                      setEmail(e.target.value.trim());
+                      setEmailExistsError('');
+                    }}
+                  />
+                  {submitted && emailExistsError && (
+                    <p className={styles.errorMessage}>{emailExistsError}</p>
```

Comment:
you can create one variables for all errors and check them inside handleSubmit function.